File: football-gui-game/src/main/java/com/github/lipinskipawel/gui/DrawableFacade.java
package com.github.lipinskipawel.gui;

import com.github.lipinskipawel.board.engine.Board;
import com.github.lipinskipawel.board.engine.Player;

final class DrawableFacade implements DrawableFootballPitch {

    private final GameDrawer gameDrawer;
    private final GamePanel gamePanel;

    DrawableFacade(final GameDrawer gameDrawer, final GamePanel gamePanel) {
        this.gameDrawer = gameDrawer;
        this.gamePanel = gamePanel;
    }

    @Override
    public void drawPitch(Board board, Player playerView) {
        this.gameDrawer.drawMove(board, playerView);
        this.gamePanel.activePlayer(board.getPlayer());
    }

    @Override
    public void activePlayer(Player activePlayer) {
        this.gamePanel.activePlayer(activePlayer);
    }
}

File: src/main/kotlin/com/github/lipinskipawel/gui/DrawableFacade.kt
package com.github.lipinskipawel.gui

import com.github.lipinskipawel.board.engine.Board
import com.github.lipinskipawel.board.engine.Player

internal class DrawableFacade(private val gameDrawer: GameDrawer, private val gamePanel: GamePanel) : DrawableFootballPitch {
    override fun drawPitch(board: Board, playerView: Player) {
        gameDrawer.drawMove(board, playerView)
        gamePanel.activePlayer(board.player)
    }

    override fun activePlayer(activePlayer: Player) {
        gamePanel.activePlayer(activePlayer)
    }
}
